aptos-move/framework/aptos-framework/sources/ResourceAccount.move

```diff
@@ -0,0 +1,130 @@
+/// A resource account is used to manage resources independent of an account managed by a user.
+/// This contains several utilities to make using resource accounts more effective.
+///
+/// A dev wishing to use resource accounts for a liquidity pool, would likely do the following:
+/// 1. Create a new account using `ResourceAccount::create_resource_account`. This creates the
+/// account, stores the `signer_cap` within a `ResourceAccount::Container`, and rotates the key to
+/// the current accounts authentication key or a provided authentication key.
+/// 2. Define the LiquidityPool module's address to be the same as the resource account.
+/// 3. Construct a ModuleBundle payload for the resource account using the authentication key used
+/// in step 1.
+/// 4. In the LiquidityPool module's `init_module` function, call `retrieve_resource_account_cap`
+/// which will retrive the `signer_cap` and rotate the resource account's authentication key to
+/// `0x0`, effectively locking it off.
+/// 5. When adding a new coin, the liquidity pool will load the capability and hence the signer to
+/// register and store new LiquidityCoin resources.
+///
+/// Code snippets to help:
+/// ```
+/// fun init_module(source: &signer) {
+///   let dev_address = @DEV_ADDR;
+///   let signer_cap = retrieve_resource_account_cap(&source, dev_address);
+///   let lp_signer = create_signer_with_capability(&signer_cap);
+///   let lp = LiquidityPoolInfo { signer_cap: signer_cap, ... };
+///   move_to(&lp_signer, lp);
+/// }
+/// ```
+///
+/// Later on during a coin registration:
+/// ```
+/// public fun add_coin<X, Y>(lp: &LP, x: Coin<x>, y: Coin<y>) {
+///     if(!exists<LiquidityCoin<X, Y>(LP::Address(lp), LiquidityCoin<X, Y>)) {
+///         let mint, burn = Coin::initialize<LiquidityCoin<X, Y>>(...);
+///         move_to(&create_signer_with_capability(&lp.cap), LiquidityCoin<X, Y>{ mint, burn });
+///     }
+///     ...
+/// }
+/// ```
+module AptosFramework::ResourceAccount {
+    use Std::Errors;
+    use Std::Signer;
+    use Std::Vector;
+    use AptosFramework::Account;
+    use AptosFramework::SimpleMap::{Self, SimpleMap};
+
+    const ECONTAINER_NOT_PUBLISHED: u64 = 0;
+
+    struct Container has key {
```

Comment:
nit: Can we have a better name than the generic Container? May be SignerCapContainer?

Reply:
It is effectively the ResourceAccount::Container. Probably a stylistic thing?